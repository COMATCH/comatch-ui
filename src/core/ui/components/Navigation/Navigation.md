
## Navigation

From [`Navigation.js`](Navigation.js)



prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**breakpoint** | `Enum('tablet','desktop','desktopLarge')` | `'desktop'` | :x: | 
**className** | `String` | `''` | :x: | 
**items** | `Array[]<ReactNode>` | `[]` | :x: | 
**orientation** | `Enum()` | `'horizontal'` | :x: | 



