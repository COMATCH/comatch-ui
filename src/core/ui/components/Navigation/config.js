export const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
    RESPONSIVE: 'responsive',
};
