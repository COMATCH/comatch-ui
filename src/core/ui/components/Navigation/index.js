export { Navigation } from './Navigation';
export { ORIENTATION } from './config';
