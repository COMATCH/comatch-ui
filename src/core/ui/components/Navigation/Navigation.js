import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { StyledWrapper } from './StyledWrapper';

export const ORIENTATION = {
    HORIZONTAL: 'horizontal',
    VERTICAL: 'vertical',
    RESPONSIVE: 'responsive',
};

const propTypes = {
    breakpoint: PropTypes.oneOf(['tablet', 'desktop', 'desktopLarge']),
    className: PropTypes.string,
    items: PropTypes.arrayOf(PropTypes.node),
    orientation: PropTypes.oneOf(Object.entries(ORIENTATION).map(([, value]) => value)),
};

const defaultProps = {
    breakpoint: 'desktop',
    className: '',
    items: [],
    orientation: ORIENTATION.HORIZONTAL,
};

export const Navigation = ({ breakpoint, className, items, orientation }) => (
    <StyledWrapper
        breakpoint={breakpoint}
        className={classNames('Navigation', className, orientation)}
        orientation={orientation}
    >
        <ul>
            {items.map((item, index) => (
                <li key={index}>{item}</li>
            ))}
        </ul>
    </StyledWrapper>
);

Navigation.propTypes = propTypes;

Navigation.defaultProps = defaultProps;
