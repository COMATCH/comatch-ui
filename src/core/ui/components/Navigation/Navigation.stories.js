import React from 'react';
import { storiesOf } from '@storybook/react';
import { Navigation } from './Navigation';

// Can be HTML nodes or plain text
const navigationItems = ['Item 1', 'Item 2', 'Item 3', 'Item 4', 'Item 5'];

storiesOf('Navigation', module)
    .add('with horizontal orientation', () => <Navigation orientation="horizontal" items={navigationItems} />)
    .add('with vertical orientation', () => <Navigation orientation="vertical" items={navigationItems} />)
    .add('with responsive orientation and tablet as the breakpoint', () => (
        <Navigation orientation="responsive" breakpoint="tablet" items={navigationItems} />
    ))
    .add('with responsive orientation and desktop as the breakpoint', () => (
        <Navigation orientation="responsive" breakpoint="desktop" items={navigationItems} />
    ))
    .add('with responsive orientation and large desktop as the breakpoint', () => (
        <Navigation orientation="responsive" breakpoint="desktopLarge" items={navigationItems} />
    ));
