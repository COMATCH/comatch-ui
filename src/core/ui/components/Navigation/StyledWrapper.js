import styled from 'styled-components';
import { device } from '../../../styles/mixins';
import { ORIENTATION } from './Navigation';

function getDisplayBasedOnOrientation(orientation, mobile = true) {
    switch (orientation) {
        case ORIENTATION.RESPONSIVE:
            return mobile ? 'block' : 'inline-block';
        case ORIENTATION.HORIZONTAL:
            return 'inline-block';
        case ORIENTATION.VERTICAL:
        default:
            return 'block';
    }
}

export const StyledWrapper = styled.nav`
    ul {
        display: block;
        margin: 0;
        padding: 0;

        li {
            display: ${({ orientation }) => getDisplayBasedOnOrientation(orientation)};
            list-style: none;
        }
    }

    &.responsive {
        ul li {
            @media ${({ breakpoint }) => device[breakpoint]} {
                display: ${({ orientation }) => getDisplayBasedOnOrientation(orientation, false)};
            }
        }
    }
`;
