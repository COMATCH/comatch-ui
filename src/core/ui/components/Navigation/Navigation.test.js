import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { Navigation } from './Navigation';
import '../../../config/tests/setup';

describe('Navigation', () => {
    it('should render the default Navigation correctly', () => {
        const navigation = renderer.create(<Navigation />).toJSON();
        expect(navigation).toMatchSnapshot();
    });

    it('should have the correct default props', () => {
        const navigation = mount(<Navigation />);
        expect(navigation.prop('className')).toEqual('');
        expect(navigation.prop('breakpoint')).toEqual('desktop');
        expect(navigation.prop('items')).toEqual([]);
        expect(navigation.prop('orientation')).toEqual('horizontal');
    });
});
