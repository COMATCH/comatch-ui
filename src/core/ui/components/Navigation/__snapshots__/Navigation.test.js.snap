// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render the default Navigation correctly 1`] = `
<nav
  className="sc-bdVaJa igpFTH Navigation horizontal"
  orientation="horizontal"
>
  <ul />
</nav>
`;
